compiler-core/src/build/project_compiler.rs

```diff
@@ -328,6 +329,156 @@ where
         }
     }
 
+    fn compile_mix_dep_package(&mut self, package: &ManifestPackage) -> Result<(), Error> {
+        let name = &package.name;
+        let mode = self.mode();
+        let target = self.target();
+        let mix_target = "prod";
+
+        // TODO: test
+        if target != Target::Erlang {
+            tracing::info!("skipping_mix_build_for_non_erlang_target");
+            return Ok(());
+        }
+
+        let build_dir = paths::build_packages(mode, target);
+        let project_dir = paths::build_deps_package(name);
+        let mix_build_dir = project_dir.join("_build").join(mix_target);
+        // Absolute build path is needed for mix to make accurate symlinks
+        let mix_build_path = std::env::current_dir()
```

Comment:
We'll need to pass in this path from the `compiler-cli` package as the `compiler-core` doesn't know how to perform side effects like interact with the file system. This is so it can be compiled to WASM

Alternatively we could add a `current_dir()` to `self.io`